Sources/WebSocketManager/Services/UserDefaultsService.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  File.swift
+//  
+//
+//  Created by Maxim on 13.09.2022.
+//
+
+
+
+import Foundation
+
+@propertyWrapper
+    struct UserDefault<T: Codable> {
+        let key: String
+        let defaultValue: T
+
+        init(_ key: String, defaultValue: T) {
+            self.key = key
+            self.defaultValue = defaultValue
+        }
+
+        var wrappedValue: T {
+            get {
+
+                if let data = UserDefaults.standard.object(forKey: key) as? Data,
+                    let user = try? JSONDecoder().decode(T.self, from: data) {
+                    return user
+
+                }
+
+                return defaultValue
+            }
+            set {
+                if let encoded = try? JSONEncoder().encode(newValue) {
+                    UserDefaults.standard.set(encoded, forKey: key)
+                }
+            }
+        }
+    }
+
+public class GlobalSettings {
+    @UserDefault("food", defaultValue: []) var food: [FoodData]
+    @UserDefault("cell", defaultValue: []) var cell: [Cell]
+    @UserDefault("tick", defaultValue: 0) var tick: Int
+    @UserDefault("lastTick", defaultValue: 0) var lastTick: Int
+    
+    func clearData() {
+        UserDefaults.standard.removeObject(forKey: "food")
+        UserDefaults.standard.removeObject(forKey: "cell")
+        UserDefaults.standard.removeObject(forKey: "tick")
+        UserDefaults.standard.removeObject(forKey: "lastTick")
+    }
+    
+    func ticksOperation(gameData: GameDataModel) {
+        if let tick = gameData.data?.tick {
+            self.tick = tick
+        }
+        
+        if let lastTick = gameData.data?.lastReceivedTick {
+            self.lastTick = lastTick
+        }
+    }
+    
+    func cellsOperation(gameData: GameDataModel) {
+
+        var cells = self.cell
+        
+        if let deeltedCells = gameData.data?.cells?.filter({ $0.del ?? false }) {
+            deeltedCells.forEach { cell in
+                cells.removeAll(where: { $0.id == cell.id })
+            }
+        }
+        
+        if let updatedCells = gameData.data?.cells?.filter({ cell in
+            (cell.isNew == false || cell.isNew == nil) &&
+            (cell.del == false ||  cell.del == nil)
+        }) {
+            updatedCells.forEach { cell in
+                if let index = cells.firstIndex(where: { cell.id == $0.id } ) {
```

Comment:
This logic is incorrect, you can't just override the previous cell object with a new one. Because as a result client will get nils again. 
You need to find oldCell and update its values with data from the new one if it is not nil.

Reply:
changed the update logic in userdefaults